### PR TITLE
Use distro.id() instead of distro.name()

### LINKED ISSUE #50844: Regression hostname module ArchLinux, Alpine, Devuan, ...
##### SUMMARY
The issue #8653 is back. The hostname module refuses to work on Arch Linux.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
hostname module
lib/ansible/module_utils/common/sys_info.py

##### ANSIBLE VERSION

```paste below
ansible 2.8.0.dev0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/user/Private/ansible-1/lib/ansible
  executable location = ./bin/ansible
  python version = 3.7.2 (default, Dec 29 2018, 21:15:15) [GCC 8.2.1 20181127]
```

##### CONFIGURATION

no output

##### OS / ENVIRONMENT

Affected:
 - Arch Linux
 - Alpine Linux
 - Probably also devuan

##### STEPS TO REPRODUCE

```bash
./bin/ansible localhost -m hostname -a name=foo
```

##### EXPECTED RESULTS

Set hostname to "foo"

##### ACTUAL RESULTS

```paste below
ansible 2.8.0.dev0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/user/Private/ansible-1/lib/ansible
  executable location = ./bin/ansible
  python version = 3.7.2 (default, Dec 29 2018, 21:15:15) [GCC 8.2.1 20181127]
Using /etc/ansible/ansible.cfg as config file
setting up inventory plugins
host_list declined parsing /etc/ansible/hosts as it did not pass it's verify_file() method
script declined parsing /etc/ansible/hosts as it did not pass it's verify_file() method
auto declined parsing /etc/ansible/hosts as it did not pass it's verify_file() method
Set default localhost to localhost
Parsed /etc/ansible/hosts inventory source with ini plugin
Loading callback plugin minimal of type stdout, v2.0 from /home/user/Private/ansible-1/lib/ansible/plugins/callback/minimal.py
META: ran handlers
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: user
<localhost> EXEC /bin/sh -c 'echo ~user && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/user/.ansible/tmp/ansible-tmp-1547384800.5708225-160500959688797 `" && echo ansible-tmp-1547384800.5708225-160500959688797="` echo /home/user/.ansible/tmp/ansible-tmp-1547384800.5708225-160500959688797 `" ) && sleep 0'
Using module file /home/user/Private/ansible-1/lib/ansible/modules/system/hostname.py
<localhost> PUT /home/user/.ansible/tmp/ansible-local-7700vpcbni_o/tmpz62somxg TO /home/user/.ansible/tmp/ansible-tmp-1547384800.5708225-160500959688797/AnsiballZ_hostname.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/user/.ansible/tmp/ansible-tmp-1547384800.5708225-160500959688797/ /home/user/.ansible/tmp/ansible-tmp-1547384800.5708225-160500959688797/AnsiballZ_hostname.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python /home/user/.ansible/tmp/ansible-tmp-1547384800.5708225-160500959688797/AnsiballZ_hostname.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /home/user/.ansible/tmp/ansible-tmp-1547384800.5708225-160500959688797/ > /dev/null 2>&1 && sleep 0'
localhost | FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "name": "foo"
        }
    },
    "msg": "hostname module cannot be used on platform Linux (Arch linux)"
}
```

##### Code

Introduced by removing the special handling here: https://github.com/ansible/ansible/commit/61b1daa65f2e1978fac127aa11f7b0831d4ff243#diff-37c256d46a757f7a359c5f4e931401f1L19

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

- `distro.id()` is a more reliable way to determine the distribution than using `distro.name()` and works with existing class name lookup methods.
- Change logic around Amazon linux parsing.
- Update tests based on new method behavior.

Fixes #50844
Closes #50847

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
`lib/ansible/module_utils/common/sys_info.py`

##### ADDITIONAL INFORMATION

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure in unit tests: https://app.shippable.com/github/ansible/ansible/runs/107340/8/tests

File: lib/ansible/module_utils/urls.py [hunk 1]
```diff
@@ -1383,7 +1383,7 @@ def fetch_url(module, url, data=None, headers=None, method=None,
         info.update(dict(msg="OK (%s bytes)" % r.headers.get('Content-Length', 'unknown'), url=r.geturl(), status=r.code))
     except NoSSLError as e:
         distribution = get_distribution()
-        if distribution is not None and distribution.lower() == 'redhat':
+        if distribution is not None and distribution.lower() == 'rhel':
```
Comment (sivel) [inline lib/ansible/module_utils/urls.py]: This seems to indicate that there may be issues with anyone else using `get_distribution` outside of us.  I feel like we might need to normalize this to match previous values.

Comment (samdoran) [reply lib/ansible/module_utils/urls.py]: `get_distribution()` now returns `Redhat` for RHEL systems.

**Comment by dakira** [comment]
This causes a regression with Sangoma-Linux. It is now not being detected anymore.

**Comment by samdoran** [comment]
@dakira Can you run a `hostname` task with Ansible 2.8 and paste the "hostname module cannot be used on platform X" result in #42726? Or open a new issue if #42726 is not related to what you are seeing.

**Pull request merged**